app/models/friendship.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal:true
+
+class Friendship < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+  validates :user_id, uniqueness: { scope: :friend_id }
```

Comment:
I don't think this does what you think it does. I am able to create two friend requests essentially. When I have already sent one the other user, the receiver can receive. This is a basic validation but based on your implementation of friendships you need more.

Se below:
![Screen Shot 2019-09-16 at 6 00 25 PM](https://user-images.githubusercontent.com/31193823/64996981-66f84800-d8ad-11e9-9b21-6109c5d107b8.png)
![Screen Shot 2019-09-16 at 6 00 39 PM](https://user-images.githubusercontent.com/31193823/64996982-66f84800-d8ad-11e9-8802-aadd88720cee.png)

What this does is prevent something like:
![Screen Shot 2019-09-16 at 6 10 30 PM](https://user-images.githubusercontent.com/31193823/64997006-737ca080-d8ad-11e9-9c58-cafcc338b889.png)

This is fine. But, based on your implementation of friendship more would need to be done. Also, at basic level, you can't allow a person to send a friend request to someone that's already sent them a friend request.

Reply:
[OPTIONAL] [STRONGLY RECOMMENDED]
What I will say is that maybe you can try using the hint provided for a milestone in handling mutual friendships. Creating an additional column upon confirming a friendship may be necessary. [OPTIONAL]
 See links mentioned in hint for milestone:
https://explainextended.com/2009/03/07/selecting-friends/
https://stackoverflow.com/questions/4219979/sql-best-practice-for-a-friendship-table

As mentioned in pathwright
![Screen Shot 2019-09-16 at 4 57 03 PM](https://user-images.githubusercontent.com/31193823/64997387-965b8480-d8ae-11e9-87f2-c3189b251641.png)